osf/metadata/rdfutils.py

```diff
@@ -0,0 +1,112 @@
+'''helpful constants and functions for working with rdf in osf
+'''
+import rdflib
+import rdflib.compare
+
+from website import settings as website_settings
+
+
+# namespace for osf's own concepts:
+OSF = rdflib.Namespace('https://osf.io/vocab/2022/')
+# namespace for resources on _this_ osf instance:
+OSFIO = rdflib.Namespace(website_settings.DOMAIN)
+# external pid namespaces:
+DOI = rdflib.Namespace('https://doi.org/')
+ORCID = rdflib.Namespace('https://orcid.org/')
+ROR = rdflib.Namespace('https://ror.org/')
+# external terminology namespaces:
+DCT = rdflib.DCTERMS
+FOAF = rdflib.FOAF
+OWL = rdflib.OWL
+RDF = rdflib.RDF
+
+
+# namespace prefixes that will be shortened by default
+# when serialized, instead of displaying the full iri
+# (in addition to rdflib's 'core' (rdf, rdfs...))
+OSF_CONTEXT = {
+    'osf': OSF,
+    'dct': DCT,
+    'foaf': FOAF,
+    'owl': OWL,
+}
+
+
+def contextualized_graph() -> rdflib.Graph:
+    '''get a new rdf graph with default namespace prefixes already bound
+    '''
+    graph = rdflib.Graph()
+    for prefix, namespace in OSF_CONTEXT.items():
+        graph.bind(prefix, namespace)
+    return graph
+
+
+def checksum_iri(checksum_algorithm, checksum_hex) -> rdflib.URIRef:
+    '''encode a checksum-based content address urn
+
+    @checksum_algorithm: str, ideally from https://www.iana.org/assignments/hash-function-text-names/
+    @checksum_hex: hexadecimal str
+    @return: rdflib.URIRef
+    '''
+    urn = f'urn:checksum:{checksum_algorithm}::{checksum_hex}'
+    return rdflib.URIRef(urn)
+
+
+def format_dct_extent(number_of_bytes: int) -> str:
+    '''format filesize value for dct:extent
+
+    following the dcterms:extent recommendation to specify in megabytes:
+    https://www.dublincore.org/specifications/dublin-core/dcmi-terms/terms/extent/
+
+    @number_of_bytes: int (filesize in bytes)
+    @return: str
+    '''
+    if number_of_bytes is None or number_of_bytes < 0:
+        return None
+    number_of_megabytes = number_of_bytes / (2**20)
+    return f'{number_of_megabytes:.6} MB'
```

Comment:
What does the `:.6` do?

Reply:
it sets the format `precision` ([the docs are obtuse](https://docs.python.org/3.11/library/string.html#grammar-token-format-spec-format_spec) -- precision is mentioned many times but that little `[.precision]` deep in `format_spec ::=` is the only place it's associated with `.`); goal was to limit excessive decimal places from division, decided on 6 so a 1-byte file would have `dct:extent "0.000001 MB"` -- buuuut now that i'm back looking at it, i think i actually wanted `:.6f` to prevent it from slipping into scientific notation (but then it'll _always_ include zeroes out to six places... blarg)

```
>>> f'{0.777777777777:.6} MB'
'0.777778 MB'
>>> f'{0.777777777777:.2} MB'
'0.78 MB'
>>> f'{2/(2**20):.6} MB'  # oops bad
'1.90735e-06'
>>> f'{2/(2**20):.6f} MB'
'0.000002'
```
will fix the add explanatory comment and update somehow to avoid scientific notation in a place it would not be helpful